### PR TITLE
remove the sharedKeys feature flag in the engine

### PR DESCRIPTION
**Pull request by paulbrauner-da**
Context: https://github.com/digital-asset/daml/issues/18240

Assumes shared=true everywhere and cleans up the code.

### CODE REVIEW

**Review by remyhaemmerle-da — APPROVED** [review]

**Review by simonmaxen-da — COMMENTED** [review]
This may cause a bit of pain for whoever is on SDK duty. The was I would go is:
1. Add a shared constructor alongside the non-shared one
1. Wait for canton to be updated with the daml version
1. Update canton to use the shared constructor
1. Remove the non-shared constructor from daml

File: canton/community/ledger/ledger-api-core/src/main/scala/com/digitalasset/canton/platform/store/packagemeta/PackageMetadata.scala [hunk 1]
```diff
@@ -50,23 +50,19 @@ object PackageMetadata {
...
-                preference = LocalPackagePreference(packageVersion, packageId),
-                allPackageIdsForName = NonEmpty(Set, packageId),
-              )
-            ),
-            packageIdVersionMap = Map(packageId -> (packageName, packageVersion)),
-          )
-      }
-    else nonUpgradablePackageMetadata
+    // Update with upgradable package metadata
+    nonUpgradablePackageMetadata.copy(
```
[the remark this answers is not in the document]
Comment (paulbrauner-da) [reply canton/community/ledger/ledger-api-core/src/main/scala/com/digitalasset/canton/platform/store/packagemeta/PackageMetadata.scala]: I have reverted the canton changes for now, but I'll keep it in mind for when I port these changes to the canton repo. Thanks!

**Comment by paulbrauner-da** [comment]
> This may cause a bit of pain for whoever is on SDK duty. The was I would go is:
> 
> 1. Add a shared constructor alongside the non-shared one

Done. Also added back a `shared` field in some error class and the `sharedKeys` flag as well: they're both being read by canton.

**Review by simonmaxen-da — APPROVED** [review]
LGTM

**Pull request merged**